common/taskvalidator/validateworkflow.go

```diff
@@ -24,62 +24,93 @@
 package taskvalidator
 
 import (
+	"context"
+
+	"go.uber.org/zap"
+
 	"github.com/uber/cadence/common/cache"
 	"github.com/uber/cadence/common/log"
 	"github.com/uber/cadence/common/log/tag"
 	"github.com/uber/cadence/common/metrics"
+	"github.com/uber/cadence/common/persistence"
+	"github.com/uber/cadence/common/reconciliation/invariant"
+	"github.com/uber/cadence/common/types"
 )
 
-// Checker is an interface for initiating the validation process.
 type Checker interface {
-	WorkflowCheckforValidation(workflowID string, domainID string, domainName string, runID string) error
+	WorkflowCheckforValidation(ctx context.Context, workflowID string, domainID string, domainName string, runID string) error
+}
+
+// Define the staleChecker interface with the methodsgit s.
+type staleChecker interface {
+	CheckAge(response *persistence.GetWorkflowExecutionResponse) (bool, error)
 }
 
-// checkerImpl is the implementation of the Checker interface.
 type checkerImpl struct {
 	logger        log.Logger
 	metricsClient metrics.Client
 	dc            cache.DomainCache
+	pr            persistence.Retryer
+	staleCheck    staleChecker
 }
 
-// NewWfChecker creates a new instance of Checker.
-func NewWfChecker(logger log.Logger, metrics metrics.Client, domainCache cache.DomainCache) Checker {
+func NewWfChecker(logger log.Logger, metrics metrics.Client, domainCache cache.DomainCache, pr persistence.Retryer) Checker {
+	zapLogger, _ := zap.NewProduction()
+	staleCheckInstance := invariant.NewStaleWorkflow(pr, domainCache, zapLogger)
+
+	staleCheck, _ := staleCheckInstance.(staleChecker)
```

Comment:
let's not ignore the cast result ok or not. if it is not ok this function should return an error